src/setfit/trainer.py

```diff
@@ -530,3 +464,52 @@ def push_to_hub(
             config,
             skip_lfs_files,
         )
+
+
+class SetFitTrainer(Trainer):
+    def __init__(
+        self,
+        model: Optional["SetFitModel"] = None,
+        train_dataset: Optional["Dataset"] = None,
+        eval_dataset: Optional["Dataset"] = None,
+        model_init: Optional[Callable[[], "SetFitModel"]] = None,
+        metric: Union[str, Callable[["Dataset", "Dataset"], Dict[str, float]]] = "accuracy",
+        loss_class=losses.CosineSimilarityLoss,
+        num_iterations: int = 20,
+        num_epochs: int = 1,
+        learning_rate: float = 2e-5,
+        batch_size: int = 16,
+        seed: int = 42,
+        column_mapping: Optional[Dict[str, str]] = None,
+        use_amp: bool = False,
+        warmup_proportion: float = 0.1,
+        distance_metric: Callable = BatchHardTripletLossDistanceFunction.cosine_distance,
+        margin: float = 0.25,
+        samples_per_label: int = 2,
+    ):
+        warnings.warn(
+            "`SetFitTrainer` has been deprecated. Please use `Trainer` instead.", DeprecationWarning, stacklevel=2
```

Comment:
Similar idea here to warn the user when it will be removed
```suggestion
            "`SetFitTrainer` has been deprecated and will be removed in v2 of SetFit. Please use `Trainer` instead.", DeprecationWarning, stacklevel=2
```

Reply:
Done!